Assignment4/alivane-23A289/Task07.py

```diff
@@ -0,0 +1,235 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# **Task 07: Querying RDF(s)**
+
+# In[75]:
+
+
+get_ipython().system('pip install rdflib')
+github_storage = "https://raw.githubusercontent.com/FacultadInformatica-LinkedData/Curso2023-2024/master/Assignment4/course_materials"
+
+
+# First let's read the RDF file
+
+# In[78]:
+
+
+from rdflib import Graph, Namespace, Literal, FOAF
+from rdflib.namespace import RDF, RDFS
+from rdflib.plugins.sparql import prepareQuery
+
+g = Graph()
+g.namespace_manager.bind('ns', Namespace("http://somewhere#"), override=False)
+g.namespace_manager.bind('vcard', Namespace("http://www.w3.org/2001/vcard-rdf/3.0#"), override=False)
+g.parse(github_storage+"/rdf/example6.rdf", format="xml")
+
+
+# **TASK 7.1: List all subclasses of "LivingThing" with RDFLib and SPARQL**
+
+# In[128]:
+
+
+# DONE
+ns = Namespace("http://somewhere#")
+
+LivingThing = ns.LivingThing
+
+print("QUERY RDFLib")
+for s, o, p in g.triples((None, RDFS.subClassOf, LivingThing)):
+    print(s)
+
+print("QUERY SPARQL")
+# Visualize the results
+q1 = prepareQuery('''
+    SELECT ?s
+    WHERE {
+        ?s RDFS:subClassOf ns:LivingThing .
+    }
+    ''',
```

Comment:
missing recursive subclass of relationships

Reply:
It is solved :)